### PR TITLE
:book:[Docs] How-to: Z-stream Automatic Updates

### LINKED ISSUE #1128: Docs: Accept automatic upgrades within a specific z-stream
Write user documentation describing how to keep automatic upgrades within a specific z-stream. If documentation already exists, review it and ensure it is still accurate and up-to-date.

Current documentation is posted up at https://operator-framework.github.io/operator-controller/
New docs should be placed in [docs/drafts](https://github.com/operator-framework/operator-controller/tree/main/docs/drafts)

### PR DESCRIPTION
**Pull request by perdasilva**
# Description

Closes #1128

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

File: docs/drafts/how-to-z-stream-upgrades.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+kind: ClusterExtension
+metadata:
+  name: argocd
+spec:
+  installNamespace: argocd
+  source:
+    sourceType: Catalog
+    catalog:
+      packageName: argocd-operator
+      version: “~2.3" # Automatically upgrade patch releases for v2.3
```
Comment (everettraven) [inline docs/drafts/how-to-z-stream-upgrades.md:18]: is `2.3.x` equivalent to this?

Should we document all the semantic ways to achieve "z-stream automatic updates"?

File: docs/drafts/how-to-z-stream-upgrades.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+Example:
+
+```yaml
+apiVersion: olm.operatorframework.io/v1alpha1
+kind: ClusterExtension
+metadata:
+  name: argocd
+spec:
+  installNamespace: argocd
```
Comment (everettraven) [inline docs/drafts/how-to-z-stream-upgrades.md:13]: There was a recent [PR](https://github.com/operator-framework/operator-controller/pull/1200) that put the `installNamespace` and `serviceAccount` fields under an `install` root field.

File: docs/drafts/how-to-z-stream-upgrades.md (hunk 1 above)
Comment (joelanford) [reply docs/drafts/how-to-z-stream-upgrades.md:18]: I don't think we need to document the synonyms for a particular version range. I think one example is sufficient here. The API reference will document all the ways.

Comment (perdasilva) [reply docs/drafts/how-to-z-stream-upgrades.md:17]: Yeah, that was my thought process as well. I think it's best to give folks one way of doing things (even if there are multiple - and document that elsewhere). Options lead to stress XD - having said that, is 2.3.x prettier?

**Review by everettraven — APPROVED** [review]

File: docs/drafts/how-to-z-stream-upgrades.md (hunk 1 above)
Comment (everettraven) [reply docs/drafts/how-to-z-stream-upgrades.md:17]: i think `~2.3` is fine. it's 1 character shorter :P

**Pull request merged**